File: x2024/python/readability/readability.py
from cs50 import get_string


def main():
    text = get_string("Text: ")
    letters = countletters(text)
    words = countwords(text)
    sentences = countsentences(text)
    L = (letters/words) * 100
    S = (sentences/words) * 100
    index = 0.0588 * L - 0.296 * S - 15.8
    if index < 1:
        print("Before Grade 1")
    elif index >= 15.5:
        print("Grade 16+")
    else:
        print(f"Grade {round(index)}")


def countletters(text):
    count = 0
    int(count)
    length = len(text)
    for i in range(0, length):
        if text[i].isalpha():
            count = count+1
    return count


def countwords(text):
    count = 0
    int(count)
    length = len(text)
    for i in range(0, length):
        if text[i].isspace():
            count = count + 1
    return count+1


def countsentences(text):
    count = 0
    int(count)
    length = len(text)
    for i in range(0, length):
        if text[i] == '.' or text[i] == '?' or text[i] == '!':
            count = count+1
    return count


main()

File: x2024/c/Readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>
int count_letters(string text);
int count_words(string text);
int count_sentences(string text);
int main(void)
{
    // Prompt the user for some text
    string text = get_string("Text: ");

    // Count the number of letters, words, and sentences in the text
    double letters = count_letters(text);
    double words = count_words(text);
    double sentences = count_sentences(text);

    // Compute the Coleman-Liau index
    double L = (letters / words) * 100;
    double S = (sentences / words) * 100;
    double index = 0.0588 * L - 0.296 * S - 15.8;

    // Print the grade level
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 15.5)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int) round(index));
    }
}
int count_letters(string text)
{
    // Return the number of letters in text
    int count = 0;

    for (int i = 0, length = strlen(text); i < length; i++)
    {
        if (isalpha(text[i]))
        {
            count++;
        }
    }
    return count;
}

int count_words(string text)
{
    // Return the number of words in text
    int count = 0;

    for (int i = 0, length = strlen(text); i < length; i++)
    {
        if (isspace(text[i]))
        {
            count++;
        }
    }
    return count + 1;
}
int count_sentences(string text)
{
    // Return the number of senetences in text
    int count = 0;

    for (int i = 0, length = strlen(text); i < length; i++)
    {
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            count++;
        }
    }
    return count;
}
